### PR TITLE
Fix broken links - November 2020

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #21204.

### CODE REVIEW

File: .openpublishing.redirection.json
Comment (gewarren) [inline .openpublishing.redirection.json:11635]: This serialization sample docs page only gets about 100 page views/month. The link to download the sample is broken and I can't find a replacement, so I deleted the article.

File: docs/core/testing/index.md
Comment (gewarren) [inline docs/core/testing/index.md:41]: Is David Coulter's tool trimming these dotnet.microsoft.com links incorrectly?

File: docs/csharp/language-reference/operators/null-forgiving.md
Comment (pkulikov) [inline docs/csharp/language-reference/operators/null-forgiving.md:48]: For this file there is already a PR #21522

**Review by BillWagner — APPROVED** [review]
This LGTM, pending the conflicts and a fresh build.

You can :shipit: once you've addressed the merge conflict.

**Pull request merged**
